### ISSUE TITLE
test_cmake_out_of_source has false positives

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#8267]
### Suggested enhancement

The `all.sh` component `test_cmake_out_of_source` checks the stderr output of `ssl-opt.sh` in order to detect errors.

However, some of them at quite innocent and not indicative of an actual issue, for example `/tests/ssl-opt.sh: 1124: kill: No such process` which sometimes happen when a process exits before the script tries to kill it, and should be ignored.

Perhaps the correct fix is to make sure that all unwanted errors ("program X does not exists") result in `ssl-opt.sh` exiting non-zero, and then we wouldn't need to look at what it writes to stderr?

**Comment by gilles-peskine-arm** [comment]
> the correct fix is to make sure that all unwanted errors ("program X does not exists") result in ssl-opt.sh exiting non-zero, and then we wouldn't need to look at what it writes to stderr?

Yes, but that's easier said than done, especially since it's only done until the next time someone adds something weird to cope with exotic configurations.

Given that this particular failure is very rare (AFAIR it's the first time I've seen it in five years), I'm tempted to do nothing. (Obviously I'd revise this opinion if it happens often.) Or, possibly, specifically hide the error from `kill`: if there's a race condition between the process dying and the script killing it, we don't actually care.

In the long term, I want to rewrite `ssl-opt.sh` in Python, making this moot.

**Comment by gilles-peskine-arm** [comment]
And also, in the long term, I want to make all `all.sh` builds out-of-tree (except for a couple that just test in-tree building), making this moot.

**Comment by mpg** [comment]
I agree this is very low priority, and may not need addressing unless this is made moot by some other change - unless of course we start seeing more of this. I just wanted it tracked so that we have a common reference if we start seeing more of this (which I hope we don't).

**Comment by gilles-peskine-arm** [comment]
This seems to be happening a little more often nowadays. The past week it's happened on two nightlies, both on FreeBSD, one on [OpenCI](https://mbedtls.trustedfirmware.org/job/mbed-tls-nightly-tests/1628/execution/node/1733/log/) and one on [internal](https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-nightly-tests/5245/execution/node/3542/log/).
